src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -171,41 +178,42 @@ def retrieve_data(
                 f"The 'read' action is not defined for the '{self.current_collection_name}' "
                 f"endpoint in {self.saas_config.fides_key}"
             )
-        self.set_saas_request_state(read_request)
 
-        # check all the values specified by param_values are provided in input_data
-        if self._missing_dataset_reference_values(
-            input_data, read_request.param_values
-        ):
-            return []
+        rows: List[Row] = []
+        for read_request in read_requests:
```

Comment:
Main change in `retrieve_data` is iterating over the `read_requests` instead of just one. The results are still aggregated to the `rows` variable and returned after all read requests have been called.